### PR TITLE
watch.go: docs on WatchResponse.Canceled

### PR DESCRIPTION
**Pull request by hongchaodeng**

### CODE REVIEW

File: clientv3/watch.go [hunk 1]
```diff
@@ -46,7 +46,8 @@ type WatchResponse struct {
 	// CompactRevision is the minimum revision the watcher may receive.
 	CompactRevision int64
 
-	// Canceled is set to indicate the channel is about to close.
+	// If the watch failed and the stream was about to close, before the channel is closed,
+	// it returns a final response which has Canceled set to true and the error.
```
Comment (heyitsanthony) [inline clientv3/watch.go]: the channel sends a final response that has Canceled set to true with a non-nil `Err()`

Comment (hongchaodeng) [inline clientv3/watch.go]: changed

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
lgtm.

**Pull request merged**